### ISSUE TITLE
Update Vet Center product with components

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#16945]
## Description
To be in compliance with the Design System and to help make sure our products receive design and accessibility updates, the Vet Center product needs to be updated to use the most recent versions of all available components.

Example pages:

- [Boston Vet Center](https://www.va.gov/boston-vet-center/)
- [St. Paul Vet Center](https://www.va.gov/st-paul-vet-center/)
- [Boston Vet Center Locations Page](https://www.va.gov/boston-vet-center/locations/)
- [St. Paul Vet Center Locations Page](https://www.va.gov/st-paul-vet-center/locations/)

### V1 components that need to be upgraded (Actionable)

**Vet Center main page template**
1. Alert
   - Currently using the v1 component but needs to be updated [to v3](https://design.va.gov/components/alert#examples---standard---v3). NOTE: This will automatically happen on 2/16 if not manually done prior.
   - NOTE: this should apply to every possible instance on the page. For example, I found an alert embedded inside the "Am I eligible for Vet Center services as a Veteran or service member?" accordion that also needs to be updated.
2. Accordion
   - Each accordion section (Prepare for your visit plus all service accordion sections) all using v1 components. These needs to be updated [to v3](https://design.va.gov/components/accordion#examples---v3). NOTE: This will automatically happen on 2/16 if not manually done prior. 

### V1 components without V3 equivalents (Icebox)
1. On this page: Currently [using v1](https://design.va.gov/components/on-this-page), which is the most up to date component
2. Back to the top: Currently [using v1](https://design.va.gov/components/back-to-top), which is the most up to date component

### Elements that are not using components that should be (Backlog)

**Vet Center main page template & Vet Center Locations template**
1. Link
   - These are currently not using a component at all but they should be using [the v1 component](https://design.va.gov/components/link/) 
   - This would also apply anywhere a basic link is used:
     - ~Get Directions~
     - View more locations
     - Inside an alert
     - Inside accordions
   - ~All "Get Directions" links (Main location, Satellite locations, Other locations) need to be updated to use [the v1 component](https://design.va.gov/components/link/)~
   - NOTE: The "Find a Vet Center location" at the bottom of the locations page has already been updated to use the correct component
2. Additional Info
   - Inside the alert, that is inside the accordion, there is an additional info element that is not using the component and should use [the v3](https://design.va.gov/components/additional-info)
3. Expandable Alert
   - Operating statuses currently are not using any component but should be using the [v1 component](https://design.va.gov/components/alert-expandable)
   - Because these are not always available, a screenshot would be helpful here:
<img width="823" alt="Screenshot 2024-01-22 at 9 35 36 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/9ef75782-e1bd-4542-a439-672dfe9f1b50">
4. Telephone
   - These are currently not using a component at all but they should be using [the v1 component](https://design.va.gov/components/telephone)
   - This would also apply not just to the main phone numbers but whenever a phone number is shown (within alerts, within accordions, etc.)
   - The main location number and the phone numbers for Satellite locations are not using a component and should be updated to use [the v1 component](https://design.va.gov/components/telephone)
   - NOTE: The numbers under "Other nearby Vet Centers" have already been updated and are using the correct component
   - We also want to make sure all exts are being parsed within the component properly

### Question
I'm not sure if Vet Centers display banners, I can't find one that has this displaying currently but that doesn't mean they don't have the option. If they do, they should be using the [v1 banner component](https://design.va.gov/components/banner/)

## Acceptance Criteria
- [x] Link component has been added to Vet Center main pages
- [x] Alert component has been updated to v3 on Vet Center main pages
- [x] Accordion component has been updated to v3 on Vet Center main pages
- [x] Telephone component has been added to Vet Center main pages
- [x] Exts in the phone numbers are bring parsed correctly
- [x] Telephone component has been added to main phone number on Vet Center location pages
- [x] Banner component has been added if this currently uses any sort of banner at all
- [ ] Requires accessibility review

## Team
Please check the team(s) that will do this work.

- [ ] CMS Team
- [ ] Public Websites
- [x] Facilities
- [ ] Accelerated Publishing

**Comment by xiongjaneg** [comment]
Proposal from Slack:
Opt out process: On 2/16, V1 components will default to V3. At that point, we'll be force upgraded unless we opt out proactively. Opting out means: modifying those existing V1 components to pass in the uswds={false}. So: opting out is still a bunch of code changes.

Opting in estimate: If we say: just opt in and verify everything looks right, the PW estimate is that this would take 1 person 1 sprint, to verify all the affected templates and make any test updates as needed.   (Chris flagged that recent component change for pagination did have test impact, so this may occur more for other components, tbd.)

Near term proposal: Rather than trying to opt in under the gun, we'd like to bias for action here and proactively opt out. Then we can schedule in opting back in, verifying behavior, and updating tests as needed.
Fran has cut tickets for opting in that we can prioritize when we're able. It would be wise to act on these sooner than later, as DS has said they plan to deprecate V1 as soon as they're able. (No estimated timeline yet, [here](https://dsva.slack.com/archives/C03R5SBELQM/p1706126998755189?thread_ts=1705503589.174489&cid=C03R5SBELQM)).

**Comment by xiongjaneg** [comment]
@xiongjaneg Need first ticket to opt out

**Comment by xiongjaneg** [comment]
Question from Laura: If we're using a v1 component but there isn't a v3 component available now, do we need to identify them now as we're doing this work? Do we put opt out on those now?

From Michelle: An audit now would be helpful. Putting putting opt out on v1 components without v3 components would reduce the risk of those being automatically changed if v3 components become available and automatically update.

@xiongjaneg to add audit to parent epic

**Comment by laflannery** [comment]
I wanted to follow up on the question above because it was answered by Amanda/Jill in slack:

Not every component is going to have a v3 equivalent when all is said an done. So we actually don't want to do a blanket opt out for that reason. We should only be opting out of v1 components that currently have an available v3 equivalent, however I will be auditing all components and their states so we will have a record of any other v1 components. That will help us going forward if the DS does say "Hey we now have v3 available for XX component" we will at least now have backlog tickets for where all those exist

**Comment by maxx1128** [comment]
I estimate this ticket at a **5** for moving a good amount of elements over to components and updating a few more. Some of them are more interactive, which moves it from a 3 to a 5 for me.

**Comment by eselkin** [comment]
So, `centralized-content.drupal.liquid` handles the `fieldCcVetCenterFaqs` field which is configured by the editors and can have any number of components that we may or may not know about, unfortunately.

But fortunately, some of the paragraph types that are loaded by the centralized-content.drupal.liquid like alert.drupal.liquid for the Boston case are already now using v3 in main.

**Comment by eselkin** [comment]
Google get directions links can't use the va-link component because:
1. cannot include sr-only text
2. doesn't respect target="_blank"

**Comment by eselkin** [comment]
Links in accordions cannot be converted to using va-link, because they are coming in from CMS as `<a>` tags. Same with some phone numbers. I would guess that some KB articles tell the editors that's how they should be putting in links and some telephone numbers.

**Comment by eselkin** [comment]
There is a problem with converting the expandable operating status to the `va-alert-expandable`
The icons used for the Vets-website component are not congruent with the color used for the notice. The exclamation circle for Notice doesn't allow for an aqua background ("info" status has an `i` in a circle).

**Comment by laflannery** [comment]
@eselkin I know we were able to add the `<va-alert-expandable>` to VBA - it was added in content-build, I also think the component itself has analytics built into it. Are we able to add this to Vet Center the same way it was added to VBA?

**Comment by xiongjaneg** [comment]
@eselkin Jordan provided the [Storybook page for this component](https://design.va.gov/storybook/?path=/docs/components-va-alert-expandable--error)

**Comment by eselkin** [comment]
@laflannery we can add the disableAnalytics attribute. But, I guess my question is: do we want to keep attempting to capture the other analytics in the way that was done on the component we have now? I'm not sure if it is really captured in prod. If we don't want analytics on the expandable, I'd be happy to take off the analytics I built extra.

**Comment by laflannery** [comment]
[Discussion with Eli here](https://dsva.slack.com/archives/C0FQSS30V/p1710178337591029), but it was decided/confirmed:
- `<va-alert-expandable>` will be used as it is on VBA, meaning:
  - Facility Notice will use [info](https://design.va.gov/storybook/?path=/docs/components-va-alert-expandable--default)
  - Facility Closed will use [error](https://design.va.gov/storybook/?path=/docs/components-va-alert-expandable--default#error) 
  - Limited service and hours will use [warning](https://design.va.gov/storybook/?path=/docs/components-va-alert-expandable--default#warning)
  - No custom analytics are needed, only the standard component analytics

**Comment by xiongjaneg** [comment]
@aklausmeier For your awareness on your return, some icon changes in Vet Center operating status

**Comment by jilladams** [comment]
From mid-sprint: Laura pointed out expandable alerts are not working in PR enviro, Eli working on tracking that down.

**Comment by eselkin** [comment]
The expandable alerts seem to work now. Laura confirmed.

**Comment by xiongjaneg** [comment]
Confirmed in staging
<img width="798" alt="Screenshot 2024-03-26 at 9 08 21 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/73648238/ee1462a6-c280-4dd0-aa81-6a9a2fa2b819">
<img width="798" alt="Screenshot 2024-03-26 at 9 09 15 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/73648238/d3341685-e0c5-49fe-86d3-b3e5781a24a4">
<img width="798" alt="Screenshot 2024-03-26 at 9 10 06 AM" src="https://github.com/department-of-veterans-affairs/va.gov-cms/assets/73648238/c2a26d81-7ceb-421e-9f9a-e9d97bee1936">

**Comment by eselkin** [comment]
Though it's interesting that they are not using the width I assigned. I think we need an issue for that, maybe. Oh wait it was a max-width. Not sure if we want to give them a fixed witdth.

**Comment by laflannery** [comment]
@eselkin FWIW, I'm seeing the operating status as wider than the screenshots above and they look good to me, pretty close to what they used to look like before the component, so I don't think we need to make any additional updates
![Screenshot 2024-03-27 at 8 32 52 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/73be1ae3-805d-4c56-9ed7-f88f8c200b8f)
